connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/aiagent/langchain4j/jobworker/L4JAiAgentJobWorkerElementTemplateRegressionTests.java

```diff
@@ -0,0 +1,125 @@
+                        .build())),
+            new ToolExecutionResultMessage("aaa111", "SuperfluxProduct", "24"),
+            new ToolExecutionResultMessage(
+                "bbb222", "Search_The_Web", "No results for 'Where does this data come from?'"),
+            new AiMessage(
+                "I played with the tools and learned that the data comes from the follow-up task and that a superflux calculation of 5 and 3 results in 24."),
+            new UserMessage("So what is a superflux calculation anyway?"),
+            new AiMessage(
+                "A very complex calculation only the superflux calculation tool can do."));
+
+    mockChatInteractions(
+        ChatInteraction.of(
+            ChatResponse.builder()
+                .metadata(
+                    ChatResponseMetadata.builder()
+                        .finishReason(FinishReason.TOOL_EXECUTION)
+                        .tokenUsage(new TokenUsage(10, 20))
+                        .build())
+                .aiMessage((AiMessage) expectedConversation.get(2))
+                .build()),
+        ChatInteraction.of(
+            ChatResponse.builder()
+                .metadata(
+                    ChatResponseMetadata.builder()
+                        .finishReason(FinishReason.STOP)
+                        .tokenUsage(new TokenUsage(100, 200))
+                        .build())
+                .aiMessage((AiMessage) expectedConversation.get(5))
+                .build(),
+            userFollowUpFeedback("So what is a superflux calculation anyway?")),
+        ChatInteraction.of(
+            ChatResponse.builder()
+                .metadata(
+                    ChatResponseMetadata.builder()
+                        .finishReason(FinishReason.STOP)
+                        .tokenUsage(new TokenUsage(11, 22))
+                        .build())
+                .aiMessage((AiMessage) expectedConversation.get(7))
+                .build(),
+            userSatisfiedFeedback()));
+
+    final var processResource = resourceLoader.getResource("classpath:regression/" + processFile);
+    final var zeebeTest =
+        deployModel(Bpmn.readModelFromStream(processResource.getInputStream()))
+            .createInstance(Map.of("userPrompt", initialUserPrompt))
+            .waitForProcessCompletion();
+
+    assertLastChatRequest(expectedConversation, false);
+
+    String expectedResponseText = ((AiMessage) expectedConversation.getLast()).text();
+    assertAgentResponse(
+        zeebeTest,
+        agentResponse ->
+            JobWorkerAgentResponseAssert.assertThat(agentResponse)
+                .isReady()
+                .hasMetrics(new AgentMetrics(3, new AgentMetrics.TokenUsage(121, 242)))
+                .hasResponseText(expectedResponseText)
+                .hasNoResponseMessage()
+                .hasNoResponseJson());
+
+    assertThat(jobWorkerCounter.get()).isEqualTo(2);
```

Comment:
🔨 nitpick: while familiarizing with the code i stumbled upon that this counter is used across both - process and task implementation tests.

We could rename this class member to sth like `executedToolTaskCounter` or sth. similar, to have a bit more clarity. WDYT?